### DISCUSSION TITLE
SetFilter <> ''

### DISCUSSION THREAD

**Discussion by Arthurvdv** [StefanMaron/BusinessCentral.LinterCop#607] [New Rule]
Create (or add to existing rule) the check for below.

**Bad code**
```AL
Customer.SetFilter("Customer Posting Group", '<>''');
```

**Good code**
```AL
Customer.SetFilter("Customer Posting Group", '<>%1', '');
```

**Comment by jwikman** [comment]
I assume that also this should be considered as **Bad code**? (since your Bad code is even Wrong code 😜)
```
Customer.SetFilter("Customer Posting Group", '<>''''');
```

But I like the idea, the "Good code" above is much more readable. 👍

**Reply by Arthurvdv** [reply]
You're absolutely right! The "bad code" was indeed ment as "this isn't going to work" 😅

I'm not a fan of the `'''''`, but not sure if there's always an solution for this. Like for example the code below


```AL
field(50100; "Stock Start Date"; Date)
{
    Caption = 'Stock Start Date';
    Editable = false;
    FieldClass = FlowField;
    CalcFormula = lookup("Warehouse Sales Item"."Stock Start Date" 
                    where(
                    "Item No." = field("No."), 
                    "Location Code" = field("Location Filter"), 
                    "Stock Start Date" = filter('<= today'), 
                    "Stock End Date" = filter('>= today|''''')
                    )
                  );
}
```

...and ignore the hardcode today, that something for (https://github.com/StefanMaron/BusinessCentral.LinterCop/discussions/593)

**Reply by jwikman** [reply]
I believe you are right, you cannot workaround ti in the CalcFormula where filter clause, at least if you want a hard coded filter (compared to a FlowFilter that is set during runtime),

**Reply by Arthurvdv** [reply]
Added a new rule for this: https://github.com/StefanMaron/BusinessCentral.LinterCop/pull/667

@jwikman, thanks again for the suggestion of `''''`, I've included this in the wiki on the rule.
